packages/docs-reanimated/docs/scroll/useAnimatedScrollHandler.mdx

```diff
@@ -4,126 +4,119 @@ sidebar_position: 3
 
 # useAnimatedScrollHandler
 
-import DocsCompatibilityInfo from '../_shared/_docs_compatibility_info.mdx';
+`useAnimatedScrollHandler` is a hook that returns an event handler reference. It can be used with React Native's scrollable components.
 
-<DocsCompatibilityInfo />
+## Reference
 
-This is a convenience hook that returns an event handler reference which can be used with React Native's scrollable components.
+```js
+import { useAnimatedScrollHandler } from 'react-native-reanimated';
+
+function App() {
+  const offsetY = useSharedValue(0);
+
+  // highlight-start
+  const scrollHandler = useAnimatedScrollHandler((event) => {
+    translationY.value = event.contentOffset.y;
+  });
+  // highlight-end
+
+  // ...
+
+  return <Animated.ScrollView onScroll={scrollHandler} />;
+}
+```
+
+<details>
+<summary>Type definitions</summary>
+
+```typescript
+interface ScrollHandlers<Context extends Record<string, unknown>> {
+  onScroll?: ScrollHandler<Context>;
+  onBeginDrag?: ScrollHandler<Context>;
+  onEndDrag?: ScrollHandler<Context>;
+  onMomentumBegin?: ScrollHandler<Context>;
+  onMomentumEnd?: ScrollHandler<Context>;
+}
+
+function useAnimatedScrollHandler<Context extends Record<string, unknown>>(
+  handlers: ScrollHandler<Context> | ScrollHandlers<Context>,
+  dependencies?: DependencyList
+): ScrollHandlerProcessed<Context>;
+
+type EventHandlerProcessed<
+  Event extends object,
+  Context extends Record<string, unknown> = never
+> = (event: Event, context?: Context) => void;
+
+type ScrollHandler<
+  Context extends Record<string, unknown> = Record<string, unknown>
+> = (event: ReanimatedScrollEvent, context: Context) => void;
+
+type ScrollHandlerProcessed<
+  Context extends Record<string, unknown> = Record<string, unknown>
+> = EventHandlerProcessed<RNNativeScrollEvent, Context>;
+```
+
+</details>
 
 ### Arguments
 
-#### `scrollHandlerOrHandlersObject` [object with worklets]
+#### `handlers`
+
+Object containing custom keys matching native event names. Following keys are available:
+
+- `onScroll`
+- `onBeginDrag`
+- `onEndDrag`
+- `onMomentumBegin`
+- `onMomentumEnd`
 
-Object containing any of the following keys: `onScroll`, `onBeginDrag`, `onEndDrag`, `onMomentumBegin`, `onMomentumEnd`.
 The values in the object should be individual worklets.
```

Comment:
There should be info that if only a single worklet function of type `(event) => void` is passed (instead of functions matched to event keys), the function is being handled as a handler for 'onScroll' event.
Reason:
<img width="690" alt="image" src="https://github.com/user-attachments/assets/ac7daf17-a796-40bc-a2bf-502f2cf73239">